### PR TITLE
Feature/138 save f3x

### PR DESCRIPTION
**Pull request by toddlees**

### CODE REVIEW

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.spec.ts [hunk 1]
```diff
@@ -52,5 +60,17 @@ describe('CreateF3XStep1Component', () => {
...
+  });
+
+  it('#save should save a new f3x record', () => {
+    component.form.patchValue({ ...f3x });
+    component.save();
+    const req = httpTestingController.expectOne(`${environment.apiUrl}/f3x-summaries/`);
+    expect(req.request.method).toEqual('POST');
+    httpTestingController.verify();
   });
 });
```
Comment (mjtravers) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.spec.ts:88]: The unit tests are generating warnings/errors about unknown elements
app-138-unit-test-errors
.

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts [hunk 2]
```diff
@@ -78,18 +83,29 @@ export class CreateF3XStep1Component implements OnInit, OnDestroy {
...
+        return of(undefined);
+      })
+    );
+    combineLatest([this.store.select(selectCommitteeAccount), existingReport$])
       .pipe(takeUntil(this.destroy$))
-      .subscribe((committeeAccount) => {
+      .subscribe(([committeeAccount, existingReport]) => {
         const filingFrequency = this.userCanSetFilingFrequency ? 'Q' : committeeAccount.filing_frequency;
         this.form.addControl('filing_frequency', new FormControl());
         this.form.addControl('report_type_category', new FormControl());
```
Comment (mjtravers) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts:107]: Why do you use addControl() here? Is it to remove the validation rules?

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts [hunk 3]
```diff
@@ -154,6 +179,37 @@ export class CreateF3XStep1Component implements OnInit, OnDestroy {
...
+      this.messageService.add({
+        severity: 'success',
+        summary: 'Successful',
+        detail: 'Contact Updated',
+        life: 3000,
+      });
+    });
+  }
+
+  public convertDateToFecFormat(date: Date | null) {
```
Comment (mjtravers) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts]: I'm thinking this function may be needed in other components as well. How about moving it to a static method in a new shared/utils/date.utils.ts class?

File: front-end/src/app/shared/pipes/fec-date.pipe.ts [hunk 4]
```diff
@@ -6,6 +6,9 @@ import { DateTime } from 'luxon';
 })
 export class FecDatePipe implements PipeTransform {
   transform(value: string): string {
+    if (!value) {
+      return '';
+    }
```
Comment (mjtravers) [inline front-end/src/app/shared/pipes/fec-date.pipe.ts:11]: Good catch and fix

**Review by mjtravers — CHANGES_REQUESTED** [review]
The ANDI and unit test checks show some errors and a couple of questions/comments. Requesting changes.

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html [hunk 5]
```diff
@@ -144,10 +144,10 @@ <h3>Covering period</h3>
   </div>
```
Comment (mjtravers) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html:144]: There are 4 issues flagged by ANDI
app-138-andi-errors

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts (hunk 2 above)
Comment (toddlees) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts:107]: these fields don't exist in the schema, but i want to keep track of them in the form

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html (hunk 5 above)
Comment (toddlees) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.html:144]: This seems to be an issue with primeng's selectButton
image

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts (hunk 2 above)
Comment (mjtravers) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts:107]: They are added to the form when FormBuilder calls in the this.formProperties file so this call may not be necessary (maybe). We can test this out in a follow up ticket

**Pull request merged**